src/org/nejrasm/zadaca4oop/task2/Rectangle.java

```diff
@@ -0,0 +1,22 @@
+package org.nejrasm.zadaca4oop.task2;
+
+public class Rectangle extends Shape {
+    private final double a;
+    private final double b;
+
+    protected Rectangle(final String colour, final double a, final double b) {
```

Comment:
You force protected too much. Yes you hide the ctor, which is generally good, but this class is not usable outside of the package  `org.nejrasm.zadaca4oop.task2;`. That is not good. Don't rely on the design choice of protected to be visible in the same package, rather make this public. It's important for attributes to be private not that much for methods.

Reply:
Noted.